tests/integration/golang/mlflow/artifact/get_artifact_s3_test.go

```diff
@@ -0,0 +1,207 @@
+//go:build integration
+
+package artifact
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/service/s3"
+	"github.com/google/uuid"
+	"github.com/hetiansu5/urlquery"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/suite"
+
+	"github.com/G-Research/fasttrackml/pkg/api/mlflow"
+	"github.com/G-Research/fasttrackml/pkg/api/mlflow/api"
+	"github.com/G-Research/fasttrackml/pkg/api/mlflow/api/request"
+	"github.com/G-Research/fasttrackml/pkg/api/mlflow/dao/models"
+	"github.com/G-Research/fasttrackml/tests/integration/golang/fixtures"
+	"github.com/G-Research/fasttrackml/tests/integration/golang/helpers"
+)
+
+type GetArtifactS3TestSuite struct {
+	suite.Suite
+	s3Client           *s3.Client
+	runFixtures        *fixtures.RunFixtures
+	serviceClient      *helpers.HttpClient
+	experimentFixtures *fixtures.ExperimentFixtures
+}
+
+func TestGetArtifactTestSuite(t *testing.T) {
+	suite.Run(t, new(GetArtifactS3TestSuite))
+}
+
+func (s *GetArtifactS3TestSuite) SetupTest() {
+	s3Client, err := helpers.NewS3Client(helpers.GetS3EndpointUri())
+	assert.Nil(s.T(), err)
+	s.s3Client = s3Client
+
+	s.serviceClient = helpers.NewMlflowApiClient(helpers.GetServiceUri())
+
+	experimentFixtures, err := fixtures.NewExperimentFixtures(helpers.GetDatabaseUri())
+	assert.Nil(s.T(), err)
+	s.experimentFixtures = experimentFixtures
+
+	runFixtures, err := fixtures.NewRunFixtures(helpers.GetDatabaseUri())
+	assert.Nil(s.T(), err)
+	s.runFixtures = runFixtures
+}
+
+func (s *GetArtifactS3TestSuite) Test_Ok() {
+	defer func() {
+		assert.Nil(s.T(), s.experimentFixtures.UnloadFixtures())
+	}()
+
+	testData := []struct {
+		name   string
+		bucket string
+	}{
+		{
+			name:   "TestWithBucket1",
+			bucket: "bucket1",
+		},
+		{
+			name:   "TestWithBucket2",
+			bucket: "bucket2",
+		},
+	}
+
+	for _, tt := range testData {
+		s.T().Run(tt.name, func(t *testing.T) {
+			// 1. create test experiment.
+			experiment, err := s.experimentFixtures.CreateExperiment(context.Background(), &models.Experiment{
+				Name: fmt.Sprintf("Test Experiment In Bucket %s", tt.bucket),
+				Tags: []models.ExperimentTag{
+					{
+						Key:   "key1",
+						Value: "value1",
+					},
+				},
+				CreationTime: sql.NullInt64{
+					Int64: time.Now().UTC().UnixMilli(),
+					Valid: true,
+				},
+				LastUpdateTime: sql.NullInt64{
+					Int64: time.Now().UTC().UnixMilli(),
+					Valid: true,
+				},
+				LifecycleStage:   models.LifecycleStageActive,
+				ArtifactLocation: fmt.Sprintf("s3://%s/1", tt.bucket),
+			})
+			assert.Nil(s.T(), err)
+
+			// 2. create test run.
```

Comment:
probably the same question - do we need to do all of that in cycle?

Reply:
removed unneeded attrs